src/js/client/controls.js

```diff
@@ -0,0 +1,150 @@
+        switch (event.keyCode) {
+            case 27: // escape
+                that.toggleMenu();
+                break;
+        }
+
+        // TODO:  Eventually, we may want to take over control of "natural" arrow key handling using event.preventDefault()
+    };
+
+    gpii.express.user.frontend.controls.handleToggleKeys = function (that, event) {
+        switch (event.keyCode) {
+            case 27: // escape
+                that.toggleMenu();
+                break;
+            case 13: // enter
+                that.toggleMenu();
+                break;
+        }
+    };
+
+    gpii.express.user.frontend.controls.handleLogoutKeys = function (that, event) {
+        switch (event.keyCode) {
+            case 13: // enter
+                that.submitForm(event);
+                break;
+        }
+    };
+
+    gpii.express.user.frontend.controls.toggleMenu = function (that) {
+        var toggle = that.locate("toggle");
+        var menu   = that.locate("menu");
+
+        if ($(menu).is(":hidden")) {
+            menu.show();
+            menu.focus();
+        }
+        else {
+            menu.hide();
+            toggle.focus();
+        }
+    };
+
+    fluid.defaults("gpii.express.user.frontend.controls", {
+        gradeNames: ["gpii.templates.templateFormControl"],
+        container:  ".controls-viewport",
+        ajaxOptions: {
+            type:     "GET",
+            url:      "/api/user/logout"
+        },
+        templates: {
+            initial: "controls-viewport",
+            success: "common-success",
+            error:   "common-error"
+        },
+        rules: {
+            successResponseToModel: {
+                "":        "notfound",
+                user: {
+                    literalValue: false
+                }
+            }
+        },
+        selectors: {
+            initial:  "",
+            success:  ".controls-message",
+            error:    ".controls-message",
+            controls: ".user-controls",
+            menu:     ".user-menu",
+            logout:   ".user-menu-logout",
+            toggle:   ".user-controls-toggle"
+        },
+        model: {
+            user: null
+        },
+        modelListeners: {
+            user: {
+                func:          "{that}.renderInitialMarkup",
+                excludeSource: "init"
+            }
+        },
+        invokers: {
+            toggleMenu: {
+                funcName: "gpii.express.user.frontend.controls.toggleMenu",
+                args:     [ "{that}"]
+            },
+            handleMenuKeys: {
+                funcName: "gpii.express.user.frontend.controls.handleMenuKeys",
+                args:     [ "{that}", "{arguments}.0"]
+            },
+            handleToggleKeys: {
+                funcName: "gpii.express.user.frontend.controls.handleToggleKeys",
+                args:     [ "{that}", "{arguments}.0"]
+            }
+        },
+        listeners: {
+            onMarkupRendered: [
```

Comment:
I split out the listeners as requested.

The "controls" component only has one thing it does itself, which is to make an AJAX call to GET /api/user/logout and then update the user in the model (setting it to null). That action is launched using the "submit" invoker of the templateFormControl grade.  This is the general job of the templateFormControl grade, to:

1. Render itself based on the initial model.
2. Submit an AJAX request when its submit invoker is called.
3. On success:
  a. Display whatever messages are configured using a particular rule.
  b. Update the module if desired based on another rule.
4. On error:
  a. Display whatever error messages are configured using a particular rule.

The templateFormControl grade replaced a bunch of forms, which is where it gots its name.  It is very similar to ajaxCapable, but adds standard wiring for success and error messages, and standard wiring for handling a "submit" action.